File: backend-py/routes/comments.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from database import get_db
from models import Comment
from auth import get_current_user_id

router = APIRouter(prefix="/api/comments")


@router.delete("/{comment_id}")
def delete_comment(comment_id: str, db: Session = Depends(get_db), user_id: str = Depends(get_current_user_id)):
    comment = db.query(Comment).filter(Comment.id == comment_id).first()
    if not comment:
        raise HTTPException(404, "Comment not found")
    if comment.author_id != user_id:
        raise HTTPException(403, "Not authorized")
    db.delete(comment)
    db.commit()
    return {"message": "Comment deleted"}

File: backend/src/routes/comments.ts
import { Router } from 'express';
import prisma from '../prisma';
import { AuthRequest, authMiddleware } from '../middleware/auth';
import { AppError } from '../middleware/error';

const router = Router();

router.delete('/:id', authMiddleware, async (req: AuthRequest, res, next) => {
  try {
    const comment = await prisma.comment.findUnique({ where: { id: req.params.id } });
    if (!comment) throw new AppError(404, 'Comment not found');
    if (comment.authorId !== req.userId) throw new AppError(403, 'Not authorized');

    await prisma.comment.delete({ where: { id: req.params.id } });
    res.json({ message: 'Comment deleted' });
  } catch (err) { next(err); }
});

export default router;
